capture/crawler_valor.py

```diff
@@ -0,0 +1,203 @@
+        message = template.format(type(ex).__name__, ex.args)
+        logger.exception(message)
+        return None
+    return title
+
+def extract_content(article):
+    """
+	Extract relevant information about news page
+    """
+
+    try:
+        body_content = article.cleaned_text
+    except Exception as ex:
+        template = "An exception of type {0} occured during extraction of news content. Arguments:\n{1!r}"
+        message = template.format(type(ex).__name__, ex.args)
+        logger.exception(message)
+        return None
+    return body_content
+
+def detect_language(text):
+    """
+	Detect the language of text using chromium_compact_language_detector
+    :param text: text to be analyzed
+    :return: {"name": portuguese, "pt"}
+    """
+    name, code, isReliable, textBytesFound, details = cld.detect(text.encode('utf8'))
+    return {"name": name, "code": code}
+
+def compress_content(html):
+    """
+    Compresses and encodes html content so that it can be BSON encoded an store in mongodb
+    :param html: original html document
+    :return: compressed an b64 encoded document
+    """
+    pickled = CP.dumps(html, CP.HIGHEST_PROTOCOL)
+    squished = zlib.compress(pickled)
+    encoded = bson.Binary(squished)  # b64.urlsafe_b64encode(squished)
+    return encoded
+
+def decompress_content(compressed_html):
+    """
+    Decompress data compressed by `compress_content`
+    :param compressed_html: compressed html document
+    :return: original html
+    """
+    # unencoded = b64.urlsafe_b64decode(str(compressed_html))
+    decompressed = zlib.decompress(compressed_html)
+    orig_html = CP.loads(decompressed)
+    return orig_html
+
+
+def download_article(url):
+	"""
+	Download the html content of a news page
+    :param url: news page's url
+    :type url: string
+    :return: news page's content
+    :rtype: requests.models.Response	
+	"""
+	article = {
+        'link': url,
+        'source': 'crawler_Valor',
+    }
+	logger.info("Downloading article: %s", url)
+	try:
+		response = requests.get(url, timeout=30)
+	except ConnectionError:
+		logger.error("Failed to fetch %s", url)
+		return
+	except Timeout:
+		logger.error("Timed out while fetching %s", url)
+		return
+
+	encoding = response.encoding if response.encoding is not None else 'utf8'
+	dec_content = response.content.decode(encoding)
+	soup = BeautifulSoup(dec_content, "lxml")    
+	extractor  = Goose({'use_meta_language': False, 'target_language':'pt'})
+	news = extractor.extract(url=url)
+
+	article['link_content'] = compress_content(dec_content)
+	article['compressed'] = True
+	article['language'] = detect_language(dec_content)
+	article['title'] =  extract_title(news)
+	article['published'] = get_published_time(soup)
+	article['main_text'] = extract_content(news)
+
+	return article
+
+if __name__ =='__main__': 
+	for url in find_articles():
+		print url
```

Comment:
I think it's a better idea to use the logger here instead of `print` statements.